CPP_STYLE_GUIDE.md

```diff
@@ -257,6 +258,21 @@ class ExampleClass {
 };
 ```
 
+### When to use AliasedBuffers
+
+When working with typed arrays that involves direct data modification
+from C++, use an AliasedBuffer when possible, because that makes it
+easier for some JS debuggers to observe these modifications.
```

Comment:
The term `some JS debuggers` reads somewhat strange to me, can we just omit the reason and direct the reader to the comment in aliased_buffer.h? (one needs to read the header to use it anyway)

Reply:
modified accordingly, and linked with the header file section for reference. Hope @addaleax is good with this change?